src/Construction/DoctrineObjectConstructor.php

```diff
@@ -63,7 +68,16 @@ public function construct(DeserializationVisitorInterface $visitor, ClassMetadat
         }
 
         // Managed entity, check for proxy load
-        if (!\is_array($data)) {
+
+        if ($data instanceof \SimpleXMLElement) {
```

Comment:
instead of creating a new array and later trying to match between $attributes and $properties, why not transform the $data (simplexml object) into a plain $data array ?

Reply:
this should make not necessary alll the changes you had to do inside the foreach loop.

Reply:
Since an Attribute and a Property could have the same Name, I decided to distinguish between them.
```
<body id="body">
  <id>12223</id>
</body>
```